src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -653,6 +653,99 @@ protected override void ApplyAnalyzerReferenceRemoved(ProjectId projectId, Analy
             }
         }
 
+        internal override void ApplyMappedFileChanges(SolutionChanges solutionChanges)
+        {
+            // After normal project changes have been applied, we need to apply changes for
+            // generated razor documents that are mapped to a location in a file outside of our workspace.
+            // Since these generated documents have CanApplyChange() == false, they will not
+            // be applied when other project changes are applied.
+            var projectChanges = solutionChanges.GetProjectChanges();
+
+            var changedMappedDocuments = new Dictionary<ProjectChanges, ImmutableArray<DocumentId>>();
+            foreach (var projectChange in projectChanges)
+            {
+                var documents = projectChange.GetChangedDocuments().Where(ShouldApplyChangesToMappedDocuments);
+                if (documents.Any())
+                {
+                    changedMappedDocuments.Add(projectChange, documents.ToImmutableArray());
```

Comment:
Not really undestanding what this extra filter is really doing here. It seems like we could just kill this entire foreach loop and just have GetMappedTextChanges directly call GetProjectChanges and loop through it and filter the same way. Otherwise it seems like this is just creating a structure to be passed through a few functions for no real need.